### PR TITLE
Remove artifact from Polygon rendering

### PR DESCRIPTION
**Pull request by hoxbro**
Fixes https://github.com/holoviz/datashader/issues/1327

image

image

Lets see how the CI handles this change.

Edit: Can still see some artifacts, if I use the holoviews example in the original issue. Can make them disappear, if I completely remove this if statement.

### CODE REVIEW

**Comment by jbednar** [comment]
Thanks for taking a stab at this! It does seem to address the zoomed-out issue:

image

But zooming in, seems like it's not actually drawing the polygons any more?

image

**Comment by jbednar** [comment]
For reference, here's what it did before the change, where the zoomed-out version has artifacts, but the zoomed-in version is fine:

image

**Comment by Hoxbro** [comment]
image

File: datashader/glyphs/polygon.py [hunk 1]
```diff
@@ -197,8 +200,10 @@ def draw_polygon(
                     # Reject edges that are above, below, or left of current pixel.
                     # Note: Edge skipped if lower vertex overlaps,
                     #       but is kept if upper vertex overlaps
-                    if (y0c >= yi or y1c < yi
-                            or (x0c < xi and x1c < xi)
+                    if (
+                        y0c >= yi
+                        or y1c < yi
+                        or (x0c < xi and x1c < xi)
```
Comment (jbednar) [inline datashader/glyphs/polygon.py:206]: Can this reformatting be reverted?

**Comment by jbednar** [comment]
Looks great! Panning and zooming with HoloViews, I can no longer detect any issues now for that dataset.  Thanks for following up on that @Hoxbro !  Can you motivate what the meaning of each of the changes is, and any risks?  

To me it seems like one risk would be if the polygon coordinates are near the `atol` value for `np.isclose`, which is 1e-08 by default.  E.g. if we were plotting nanometer-scale shapes on an integrated circuit wafer, would we have problems from the shape coordinates being considered "close"? Or are all such comparisons being done in pixel space, where 1e-08 is indeed a very small tolerance?

**Comment by Hoxbro** [comment]
I have removed `np.isclose`, which I played around with when trying to find the problem. The coordinates is in pixel space, but I saw no difference with it. In theory, it could catch some of the cross-products, which are close to zero, but I don't feel it is worth the extra calculations for a case that right now is only theoretical. 

For risk related to the removal of  `+1` I can't say as this is pretty deep in the machinery combined with some theory I haven't dove deep into.

**Review by jbednar — APPROVED** [review]
Thanks for this. In the end, it's a very simple change, and seems to address the underlying issue (starting to draw one pixel too late).  I've studied the test cases that changed and I _think_ they make sense. The extent that's being drawn has changed slightly, but that seems to be what's required to fix the bug.  So as far as I can see, this deals with the reported bug. Thanks again!

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -66,16 +66,26 @@ jobs:
...
           MATRIX=$(jq -nsc '{
               "os": ["ubuntu-latest", "macos-latest", "windows-latest"],
-              "python-version": ["3.9", "3.10", "3.11", "3.12"]
+              "python-version": ["3.9", "3.10", "3.11", "3.12"],
+              "exclude": [{
+                  "os": "windows-latest",
+                  "python-version": "3.9"
+              }]
           }')
+          # Windows won't solve on Windows + Python 3.9
```
Comment (hoxbro) [inline .github/workflows/test.yaml:75]:
```suggestion
          # Won't solve on Windows + Python 3.9
```

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -66,16 +66,26 @@ jobs:
...
```
Comment (hoxbro) [inline .github/workflows/test.yaml:88]:
```suggestion
          # Won't solve on Windows + Python 3.9
```

**Pull request merged**